src/ray/gcs/task_info_accessor.h

```diff
@@ -0,0 +1,77 @@
+#ifndef RAY_GCS_TASK_INFO_ACCESSOR_H
+#define RAY_GCS_TASK_INFO_ACCESSOR_H
+
+#include "ray/common/id.h"
+#include "ray/gcs/callback.h"
+#include "ray/gcs/subscription_executor.h"
+#include "ray/gcs/tables.h"
+
+namespace ray {
+
+namespace gcs {
+
+class RedisGcsClient;
+
+/// \class TaskInfoAccessor
+/// `TaskInfoAccessor` is a sub-interface of `GcsClient`.
+/// This class includes all the methods that are related to accessing
+/// task information in the GCS.
+class TaskInfoAccessor {
+ public:
+  virtual ~TaskInfoAccessor() {}
+
+  /// Add a task to GCS asynchronously.
+  ///
+  /// \param data_ptr The task that will be added to GCS.
+  /// \param callback Callback that will be called after task has been added
+  /// to GCS.
+  /// \return Status
+  virtual Status AsyncAdd(const std::shared_ptr<TaskTableData> &data_ptr,
+                          const StatusCallback &callback) = 0;
+
+  /// Get task information from GCS asynchronously.
+  ///
+  /// \param task_id The ID of the task to look up in GCS.
+  /// \param callback Callback that is called after lookup finished.
+  /// \return Status
+  virtual Status AsyncGet(const TaskID &task_id,
+                          const OptionalItemCallback<TaskTableData> &callback) = 0;
+
+  /// Delete tasks from GCS asynchronously.
+  ///
+  /// \param task_ids The vector of IDs to delete from GCS.
+  /// \param callback Callback that is called after delete finished.
+  /// \return Status
+  // TODO(micafan) Will support callback of batch deletion in the future.
+  // Currently this callback will never be called.
+  virtual Status AsyncDelete(const std::vector<TaskID> &task_ids,
+                             const StatusCallback &callback) = 0;
+
+  /// Subscribe to any update operations of a task from GCS asynchronously.
```

Comment:
```suggestion
  /// Subscribe asynchronously to the event that the given task is added in GCS.
```
Because tasks are immutable, there's no update.

Reply:
done